src/OpenTelemetry.Api.ProviderBuilderExtensions/Trace/OpenTelemetryDependencyInjectionTracingServiceCollectionExtensions.cs

```diff
@@ -50,20 +48,56 @@ public static class OpenTelemetryDependencyInjectionTracingServiceCollectionExte
     /// can be chained.</returns>
     public static IServiceCollection ConfigureOpenTelemetryTracerProvider(
         this IServiceCollection services,
-        Action<IServiceProvider, TracerProviderBuilder> configure)
+        Action<TracerProviderBuilder> configure)
     {
-        RegisterBuildAction(services, configure);
+        Guard.ThrowIfNull(services);
+        Guard.ThrowIfNull(configure);
+
+        configure(new TracerProviderServiceCollectionBuilder(services));
 
         return services;
     }
 
-    private static void RegisterBuildAction(IServiceCollection services, Action<IServiceProvider, TracerProviderBuilder> configure)
+    /// <summary>
+    /// Registers an action used to configure the OpenTelemetry <see
+    /// cref="TracerProviderBuilder"/> once the <see cref="IServiceProvider"/>
+    /// is available.
+    /// </summary>
+    /// <remarks>
+    /// Notes:
+    /// <list type="bullet">
+    /// <item>This is safe to be called multiple times and by library authors.
+    /// Each registered configuration action will be applied
+    /// sequentially.</item>
+    /// <item>A <see cref="TracerProvider"/> will NOT be created automatically
+    /// using this method. To begin collecting metrics use the
+    /// <c>IServiceCollection.AddOpenTelemetry</c> extension in the
+    /// <c>OpenTelemetry.Extensions.Hosting</c> package.</item>
+    /// <item>The supplied configuration delegate is called once the <see
+    /// cref="IServiceProvider"/> is available. Services may NOT be added to a
+    /// <see cref="TracerProviderBuilder"/> once the <see
+    /// cref="IServiceProvider"/> has been created. Many helper extensions
+    /// register services and may throw if invoked inside the configuration
+    /// delegate.</item>
+    /// </list>
+    /// </remarks>
+    /// <param name="services">The <see cref="IServiceCollection" /> to add
+    /// services to.</param>
```

Comment:
Removing "to add services to" since that can't be done when using this overload right?

```suggestion
    /// <param name="services">The <see cref="IServiceCollection" />.</param>
```

Reply:
Updated